core/src/main/kotlin/com/ritense/valtimo/logging/UserLoggingFilter.kt

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2015-2024 Ritense BV, the Netherlands.
+ *
+ * Licensed under EUPL, Version 1.2 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://joinup.ec.europa.eu/collection/eupl/eupl-text-eupl-12
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" basis,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.ritense.valtimo.logging
+
+import com.ritense.valtimo.contract.LoggingConstants
+import com.ritense.valtimo.contract.authentication.UserManagementService
+import jakarta.servlet.FilterChain
+import jakarta.servlet.http.HttpServletRequest
+import jakarta.servlet.http.HttpServletResponse
+import org.slf4j.MDC
+import org.springframework.web.filter.OncePerRequestFilter
+import java.util.UUID
+
+class UserLoggingFilter(
+    private val userManagementService: UserManagementService
+) : OncePerRequestFilter() {
+
+    override fun doFilterInternal(
+        request: HttpServletRequest,
+        response: HttpServletResponse,
+        filterChain: FilterChain
+    ) {
+        val userIdentifier = userManagementService.getCurrentUser()?.getUserIdentifier()
```

Comment:
For every API call, a request to Keycloak is made. Making it a bit expensive. Is it possible to get the userIdentifier from the JWT token?

Reply:
Currently this is unfortunately not possible to do differently without a breaking change. We should look into improving this soon.

Reply:
https://github.com/valtimo-platform/valtimo-backend-libraries/pull/1544